script.js

```diff
@@ -12,55 +14,67 @@ thead.appendChild(trHead);
 table.appendChild(tbody);
 
 
+class Films {
+    constructor(id, name, genre, assessment) {
+        this._id = id;
+        this._name = name.trim();
+        this._genre = genre.trim();
+        this._assessment = assessment;
+    }
+//id
+    set id(id) {
+        this._id = id
+    }
+
+    get id() {
+        return this._id;
+    }
+//name
+    set name(name) {
+        this._name = name.trim();
+    }
+    get name() {
+        return this._name;
+    }
+    //genre
+    set genre(genre) {
+        this._genre = genre
+    }
+
+    get genre() {
+        return this._genre;
+    }
+//assessment
+    set assessment(assessment) {
+        this._assessment = assessment
+    }
+
+    get assessment() {
+        return this._assessment;
+    }
+}
+
+
+class Film extends Films {
+    constructor(id, name, genre, releaseDate, countries, assessment) {
+        super(id, name, genre, assessment);
+        this.releaseDate = releaseDate.trim();
+        this.countries = countries.trim();
+    }
+}
+
+// проверка set и get
+// const test = new Films(1, 'Papillon', 'Drama, crime', 'September 9, 2017', 'USA', 8.9);
+// test.name = "Miron";
+// console.log(test.name);
+
 const films = [
-    {
-        id: 1,
-        name: 'Papillon',
-        genre: 'Drama, crime',
-        releaseDate: 'September 9, 2017',
-        countries: 'USA',
-        assessment: 8.9
-    },
-    {
-        id: 2,
-        name: 'The Hangover',
-        genre: 'Сomedy',
-        releaseDate: 'May 30, 2009 ',
-        countries: 'USA',
-        assessment: 9.0
-    },
-    {
-        id: 3,
-        name: 'Babysitting',
-        genre: 'Сomedy',
-        releaseDate: '16 April 2014',
-        countries: 'France',
-        assessment: 8.7
-    },
-    {
-        id: 4,
-        name: 'Venom',
-        genre: 'Drama, crime',
-        releaseDate: 'October 1, 2018',
-        countries: 'USA',
-        assessment: 7.8
-    },
-    {
-        id: 5,
-        name: 'Papillon',
-        genre: 'Superhero movie, crime',
-        releaseDate: 'September 9, 2017',
-        countries: 'USA',
-        assessment: 8.5
-    },
-    {
-        id: 6,
-        name: 'Papillon',
-        genre: 'Drama, crime',
-        releaseDate: 'September 9, 2017',
-        countries: 'USA',
-        assessment: 8.5
-    },
```

Comment:
I would leave this as it is and separately use 'forEach' to build 'objects array'